### DISCUSSION TITLE
Fusion 360 post processor for 4th axis errors.

### DISCUSSION THREAD

**Discussion by dalewooders** [grblHAL/PostProcessors#6] [Q&A]
Please excuse my very limited knowledge on this.  

I've recently added a 4th rotary axis in line with the Y on a CNC router. I originally tried to get it running with the included grbl post process but from my very very limited knowledge of the code it doesn't look like a 4th axis is supported.[https://github.com/Dietz0r/grblHAL_Fusion360_Post_Processor](https://github.com/Dietz0r/grblHAL_Fusion360_Post_Processor?fbclid=IwAR18oYlHzhZRLhoWHytkHT9cg6dfQT43yLIsNKEaVmUC6Ue52kqt3gcBi2U) looks like it should work. I've even built the machine in machine builder and used the machine configuration to try and set up the post.
I'm using tool orientation to index the rotation in fusion but I keep getting errors. WCS is set up alonmg the centreline of the y axis. Simulation runs fine. Anyone with any bright ideas? I feel like I'm close but it's driving me nuts! 
![Screenshot 2024-03-03 123135](https://github.com/grblHAL/PostProcessors/assets/162236858/64509b0a-036b-4281-b3b9-baa6c0d74f05)

**Comment by Toutoune86** [comment]
Hello,
I am not good at coding either, so I am not sure I will be able to help you a lot. Yet, I have two suggestions, and I may be interested in whatever solutions you find as well (I have a fourth axis waiting to be used too ; but I may use SheetCAM rather than Fusion).

1st suggestion (may be not a prority): If your rotary axis is parallel to Y, it wonder if it should be defined as B axis, not A, both on your breakout board wiring to motor driver and some grbl setup. Actually I will be interested to know what would be necessary in that sense, cause I remember having read somewhere the main CNC convention is to have rotary axis parallel to X named A, and those parallel to Y called B.

2nd suggestion : 
It looks like the rotary axis is to befined at the beginning of the postprocessor, from line 162. How to define, what to write ? I am not sure to understand yet properly, but it looks necessary.
![Capture postpro 4ème axe](https://github.com/grblHAL/PostProcessors/assets/27381616/4aaa36c6-cff1-46b9-9303-0aee15f399a8)

**Comment by Toutoune86** [comment]
I just thought about it  again. Just realize this thing can be set directly from fusion postprocessor setting menu, no need to edit directly the PP file. Sorry
